filter.cc

```diff
@@ -164,92 +180,48 @@ void Filter::swapData(Filter *other) {
   other->domainList = tempDomainList;
   other->host = tempHost;
   other->hostLen = tempHostLen;
+  other->domainsParsed = tempDomainsParsed;
+  other->domains = tempDomains;
+  other->antiDomains = tempAntiDomains;
 }
 
-bool isDomain(const char *input, int len, const char *domain, bool anti) {
-  const char *p = input;
-  if (anti) {
-    if (len >= 1 && p[0] != '~') {
+bool Filter::containsDomain(const char* domain, size_t domainLen,
```

Comment:
See the header file for a description of this. Mainly it just checks if the specified domain is inside the filter's list of domains (or anti domains).